examples/stationary-heat-equation.cc

```diff
@@ -79,15 +83,21 @@ int main(int argc, char* argv[])
           return result;
         });
 
+    // Both Neumann and Dirichlet boundary conditions
     auto grid = XT::Grid::make_cube_grid<G>(/*lower_left=*/0., /*upper_right=*/1., /*num_elements=*/128);
-    auto grid_view = grid.leaf_view();
-
     XT::Grid::NormalBasedBoundaryInfo<I> boundary_info;
     boundary_info.register_new_normal({0., -1.}, new XT::Grid::NeumannBoundary());
-    boundary_info.register_new_normal({0., 1.}, new XT::Grid::NeumannBoundary());
-    boundary_info.register_new_normal({-1., 0.}, new XT::Grid::DirichletBoundary());
+    boundary_info.register_new_normal({0., 1.}, new XT::Grid::DirichletBoundary());
+    boundary_info.register_new_normal({-1., 0.}, new XT::Grid::NeumannBoundary());
     boundary_info.register_new_normal({1., 0.}, new XT::Grid::DirichletBoundary());
 
+    // Only Dirichlet boundary conditions, in that case we need to change the domain to [-1,1]^2 to ensure correct
+    // boundary conditions
+    // auto grid = XT::Grid::make_cube_grid<G>(/*lower_left=*/-1., /*upper_right=*/1., /*num_elements=*/128);
```

Comment:
Sorry for that, I also have related changes locally. What is true is that we need the `[-1, 1]^2` cube here for the exact solution and the boundary values to make sense. For the same reason, we should not treat Neumann boundary values here at all. So I would propose to only have these lines here:
```c++
auto grid = XT::Grid::make_cube_grid<G>(/*lower_left=*/-1., /*upper_right=*/1., /*num_elements=*/128);
auto grid_view = grid.leaf_view();

const XT::Grid::AllDirichletBoundaryInfo<I> boundary_info;
```

Reply:
I think the the domain [0,1]^2 with Neumann boundary conditions on the left and bottom side also makes sense, as the gradient of the exact solution is zero on these boundaries. But I am happy with changing the file as you suggested.